// react-native-notifications/.history/lib/src/notification.ios_20190708023235.js
export default class IOSNotification {
  _data: Object;
  _alert: string | Object;
  _sound: string;
  _badge: number;
  _category: string;
  _type: string; // regular / managed
  _thread: string;

  constructor(notification: Object) {
    this._data = {};

    if (notification.data.aps &&
      notification.data.aps['content-available'] &&
      notification.data.aps['content-available'] === 1 &&
      !notification.data.aps.alert &&
      !notification.data.aps.sound &&
      notification.managedAps) {
      // managed notification
      this._alert = notification.managedAps.alert;
      this._sound = notification.managedAps.sound;
      this._badge = notification.data.aps.badge;
      this._category = notification.managedAps.category;
      this._type = 'managed';
      this._thread = notification.data.aps['thread-id'];
    } else if (
      notification.data.aps &&
      notification.data.aps.alert) {
      // regular notification
      this._alert = notification.data.aps.alert;
      this._sound = notification.data.aps.sound;
      this._badge = notification.data.aps.badge;
      this._category = notification.category;
      this._type = 'regular';
      this._thread = notification.thread;
    }

    Object.keys(notification).filter(key => key !== 'data').forEach(key => {
      this._data[key] = notification[key];
    });
  }

  getMessage(): ?string | ?Object {
    return this._alert;
  }

  getSound(): ?string {
    return this._sound;
  }

  getBadgeCount(): ?number {
    return this._badge;
  }

  getCategory(): ?string {
    return this._category;
  }

  getData(): ?Object {
    return this._data;
  }

  getType(): ?string {
    return this._type;
  }

  getThread(): ?string {
    return this._thread;
  }
}
